### ISSUE TITLE
LID-01 Node Operator Rewards Unevenly Leaked 

### ISSUE THREAD

**Issue by ujenjt** [lidofinance/lido-dao#237]
Node operators get part of the fees as reward. This reward is proportional to the number of validators the node operator has. The calculation happens in `NodeOperatorsRegistry.distributeRewards` method. Current ([`v0.2.1-rc.0`](https://github.com/lidofinance/lido-dao/tree/v0.2.1-rc.0)) implementation uses integer division for every node operator, and invariably results in some small remainder that stays with the NodeOperatorRegistry contract balance. As the `distributeRewards` method distributes only current reward amount and not the whole balance, those remainders accumulate.

The audit proposes to design a more sophisticated process for reward distribution. The points are:
* It’s better to distribute the whole NodeOperatorsRegistry token balance, preventing remainders from accumulating.
* Calculate the operators’ rewards using single division — calculate the amount for one validator and distribute fees accordingly.

**Comment by skozin** [comment]
I still don't think that the leakage is uneven; let me explain why.

The leakage stems from the fact that each node operator receives `operator.effectiveStake * totalReward / totalEffectiveStake`, where the division is an integer one so it can result in the precision loss, and this precision loss can be different depending on the number of operator's active validators (effective stake).

But this loss for operators with each distinctive number of validators also depends on the value of `totalReward`. Since the latter can be considered a random variable, the precision loss will also be a random variable parametrized by the number of operator's validators, so in the long run, the leakage should still be even.

**Comment by ujenjt** [comment]
@skozin yes, you are right, but I've implemented per validator reward calculation just because it's more clear in this case.
```sol
uint256 perValidatorReward = _totalRewardShares.div(effectiveStakeTotal);

for (idx = 0; idx < activeCount; ++idx) {
    shares[idx] = operator.effectiveStake.mul(perValidatorReward);
}
```

**Comment by skozin** [comment]
Fixed in #236 as per the Eugene's comment above.
